src/mesh_processor/src/generate_mesh.cpp

```diff
@@ -0,0 +1,160 @@
+#include "generate_mesh.h"
+
+#include <boost/range/algorithm.hpp>
+#include <cinolib/find_intersections.h>
+#include <cinolib/profiler.h>
+#include <cinolib/tetgen_wrap.h>
+
+#include <execution>
+
+#include "utils.h"
+
+namespace {
+bool operator>(const cinolib::vec3d& lhs, const cinolib::vec3d& rhs) {
+	return std::tie(lhs.x(), lhs.y(), lhs.z()) >
+	       std::tie(rhs.x(), rhs.y(), rhs.z());
+}
+}  // namespace
+
+namespace MeshProcessor {
+using namespace cinolib;
+
+CustomTetmesh MeshGenerator::get_tet_mesh() {
+	Profiler profiler;
+	profiler.push("mesh creation");
+
+	auto tet_meshes = generate_layers_meshes(representation_.layers());
+	auto heat_data  = representation_.shapes_metadata();
+	assert(heat_data.size() == tet_meshes.size() &&
+	       "Heat data and shapes mismatch");
+	CustomTetmesh total_tetmesh;
+	for (auto i = 0; i < tet_meshes.size(); ++i) {
+		const auto shape_heat_data = heat_data[i];
+		auto& tet_mesh             = tet_meshes[i];
```

Comment:
please replace with heat_data.at(i); to align it with const

Reply:
done